common/lib/client/auth.js

```diff
@@ -534,6 +519,144 @@ var Auth = (function() {
 		}
 	};
 
+	Auth.prototype._saveBasicOptions = function(defaultTokenParams, authOptions) {
+		this.method = 'basic';
+		this.key = authOptions.key;
+		this.basicKey = toBase64(authOptions.key);
+		this.tokenParams = defaultTokenParams || {};
+		this.authOptions = authOptions || {};
+		delete this.authOptions.force;
+		if('clientId' in authOptions) {
+			this._userSetClientId(authOptions.clientId);
+		}
+	}
+
+	Auth.prototype._saveTokenOptions = function(tokenParams, authOptions) {
+		this.method = 'token';
+
+		this.tokenParams = tokenParams || this.tokenParams || {};
+
+		/* If an authOptions object is passed in that contains new auth info (ie
+		* isn't just {force: true} or something), it becomes the new default, with
+		* the exception of the force attribute (RSA10g), which is set anew on each
+		* call to authorise (defaulting to false) */
+		this.force = false;
+		if(authOptions) {
+			this.force = authOptions.force;
+
+			if(containsAuthInfo(authOptions)) {
+				this.authOptions = authOptions || {};
```

Comment:
I dunno, merging seems unintuitive to me. Especially in the case of authOptions: say someone initializes the library with an authCallback. then they call authorise() with an authUrl. should the library really keep using the authCallback just because it now has both an authCallback happens to come first in the elfsif logic?